### PR TITLE
Add test coverage for JsonApiClient::Errors::ConnectionError

### PR DESCRIPTION
**Pull request by tvararu**
### Context

I added a rescue for this error in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/2031e8e385c7cc0e8b7caaf13a04248203fbb4be/app/models/find_api/course.rb#L18, but couldn't figure out how to test for it. Duncan pointed me in the right direction, so I've added in a test.

### Changes proposed in this pull request

Adds more tests for an edge case.

### Guidance to review

I used another `shared_examples` to DRY up the specs, but I'm not sure if this is the best option.

### CODE REVIEW

File: spec/system/candidate_interface/candidate_applying_spec.rb [hunk 1]
```diff
@@ -92,6 +90,18 @@
     end
   end
 
+  context 'when Find is returning server errors' do
+    let(:find_api_request) { stub_503_from_find(provider_code, course_code) }
+
+    include_examples 'falls back to basic version'
```
Comment (duncanjbrown) [inline spec/system/candidate_interface/candidate_applying_spec.rb]: 👌

**Comment by duncanjbrown** [comment]
💯
